src/main/java/com/onaio/steps/activities/NewHouseholdActivity.java

```diff
@@ -72,20 +85,45 @@ private void populateDataFromIntent() {
     }
 
     public void doneBtnClicked(View view) {
+        if (!progressDialog.isShowing()) {
+            progressDialog.show();
+        }
+
+        saveHousehold();
+    }
+
+    public void cancel(View view){
+        finish();
+    }
+
+    public void saveHousehold() {
         try {
+            ODKBlankForm odkBlankForm = (ODKBlankForm) ODKBlankForm.find(this, getFormId());
             Intent intent = this.getIntent();
             Household household = new HouseholdViewWrapper(this).getHousehold(R.id.generated_household_id, R.id.household_number,R.id.household_comments);
+            household.setOdkJrFormId(odkBlankForm.getJrFormId());
+            household.setOdkJrFormTitle(odkBlankForm.getDisplayName());
             household.save(db);
             intent.putExtra(Constants.HH_HOUSEHOLD,household);
             setResult(RESULT_OK, intent);
+            progressDialog.dismiss();
             finish();
         } catch (Exception e) {
-            new CustomDialog().notify(this, CustomDialog.EmptyListener,e.getMessage(),R.string.error_title);
+            exceptionHandler.handle(e);
         }
     }
 
-    public void cancel(View view){
-        finish();
+    public String getFormId() {
+        return KeyValueStoreFactory.instance(this).getString(Constants.HH_FORM_ID);
     }
 
+    @Override
+    public void tryToResolve() {
+        saveHousehold();
+    }
+
+    @Override
+    public void onDismiss(Exception e, int message) {
+        progressDialog.dismiss();
```

Comment:
I think we need to remove the dialog before we show the `CustomDialog`?

Reply:
done